src/main/java/org/gradle/profiler/ScenarioLoader.java

```diff
@@ -212,80 +218,26 @@ public ScenarioLoader(GradleBuildConfigurationReader gradleBuildConfigurationRea
             String title = scenario.hasPath(TITLE) ? scenario.getString(TITLE) : null;
 
             List<BuildMutator> mutators = BUILD_MUTATOR_CONFIGURATORS.entrySet().stream()
-                .filter(entry -> scenario.hasPath(entry.getKey()))
```

Comment:
filtering isn't done here. The BuildMutator's should be responsible for determining if they support the configuration or not.